integration/spark/shared/src/main/java/io/openlineage/spark/api/Vendors.java

```diff
@@ -0,0 +1,51 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.api;
+
+import io.openlineage.spark.agent.lifecycle.VisitorFactory;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public interface Vendors {
+
+  static Vendors getVendors() {
+    ServiceLoader<Vendor> serviceLoader = ServiceLoader.load(Vendor.class);
+    List<Vendor> vendors =
+        StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(
+                    serviceLoader.iterator(), Spliterator.IMMUTABLE & Spliterator.DISTINCT),
+                false)
+            .filter(Vendor::isVendorAvailable)
+            .collect(Collectors.toList());
+
+    return new VendorsImpl(vendors);
```

Comment:
Would be nice to provide test for this mechanism similar to https://github.com/OpenLineage/OpenLineage/blob/31f8ce588526e9c7c4bc7d849699cb7ce2969c8f/integration/spark/app/src/test/java/io/openlineage/spark/agent/util/TestOpenLineageEventHandlerFactory.java#L33 which is also loaded by `ServiceLoader`

Reply:
``I add a test for the Vendors interface https://github.com/OpenLineage/OpenLineage/pull/2405/commits/c37134f6375f28dd0eeffa584e3fa2dc4490ffb6 
Let me know if it's enough, if want to test also this part, I need to be a little bit more creative. 

`io.openlineage.spark.agent.lifecycle.ContextFactory` 
https://github.com/OpenLineage/OpenLineage/pull/2405/commits/304deb2b47a4f970fccb4ee3a2f9a19dc8c8875ca
```java
OpenLineageContext.builder()
    .sparkSession(Optional.of(sparkSession))
    .sparkContext(sparkSession.sparkContext())
    .openLineage(new OpenLineage(Versions.OPEN_LINEAGE_PRODUCER_URI))
    .queryExecution(queryExecution)
    .customEnvironmentVariables(
        this.openLineageEventEmitter.getCustomEnvironmentVariables())
    .vendors(Vendors.getVendors())
    .build();
```